docs/enhancements/agent-upgrade.md

```diff
@@ -0,0 +1,189 @@
+---
+title: agent-upgrade
+authors:
+  - "@jhernand"
+creation-date: 2022-06-13
+last-updated: 2022-06-13
+---
+
+# Agent upgrade
+
+## Summary
+
+Usually the time since the agent is started and the cluster is installed is
+short. But in some situations it may be long, even many days, and during that
+long time the service may be upgraded and require a newer version of the agent.
+The current mechanism to perform that agent upgrade is to reboot the host. That
+is inconvenient, specially if there are many hosts or if rebooting a host
+requires manual intervention. This is a proposal to implement a mechanism to
+upgrade the agent without having to reboot the host.
+
+## Motivation
+
+Reduce the need to reboot hosts when there are changes in the service that
+require new versions of the agent.
+
+### Goals
+
+A host that has been booted and registered with matching versions of the agent
+and the service will not need to be rebooted in order to install the cluster
+with a newer version of the service.
+
+Note that in this context _version_ is really a container image name, like
+`registry-proxy.engineering.redhat.com/rh-osbs/openshift4-assisted-installer-agent-rhel8:v1.0.0-141`.
+
+### Non-Goals
+
```

Comment:
Worth mentioning that a change of [INSTALLER_IMAGE or CONTROLLER_IMAGE ](https://github.com/openshift/assisted-service/blob/c1eccd19b6f7a6bdd94639f074a675f6b4a43fcc/internal/host/hostcommands/instruction_manager.go#L54-L55) during installation will not reconciled like the AGENT_IMAGE will be after this enhancement.

The implication of this is that service<->installer and service<->controller communication (which is mostly one directional, aimed at the service) has to remain backwards compatible

Reply:
I don't think the installer image and controller image have something to do with this enhancement.

Reply:
Added a note explaining that upgrading those images is not related and out of scope.